### PR TITLE
[SCB-1391] Add support for explicit tx context passing for TCC

### PR DESCRIPTION
**Pull request by chanjarster**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: omega/omega-transaction/src/main/java/org/apache/servicecomb/pack/omega/transaction/TransactionContextUtils.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package org.apache.servicecomb.pack.omega.transaction;
+
+import org.apache.servicecomb.pack.omega.context.OmegaContext;
+import org.apache.servicecomb.pack.omega.context.TransactionContext;
+import org.apache.servicecomb.pack.omega.context.TransactionContextProperties;
+import org.slf4j.Logger;
+
+public abstract class TransactionContextUtils {
```
Comment (WillemJiang) [inline omega/omega-transaction/src/main/java/org/apache/servicecomb/pack/omega/transaction/TransactionContextUtils.java]: Static method is hard for extension, my suggestion is letting the TransactionAspect, TccParticipatorAspect extends the class of TransactionContextHelper.

File: omega/omega-transaction/src/test/java/org/apache/servicecomb/pack/omega/transaction/TransactionAspectTest.java [hunk 2]
```diff
@@ -17,47 +17,43 @@
 
 package org.apache.servicecomb.pack.omega.transaction;
```
Comment (WillemJiang) [inline omega/omega-transaction/src/test/java/org/apache/servicecomb/pack/omega/transaction/TransactionAspectTest.java]: Please use the google-style which could find how to setup the developement environment [here](http://servicecomb.apache.org/developers/setup-develop-environment/)

File: omega/omega-transaction/src/test/java/org/apache/servicecomb/pack/omega/transaction/TransactionContextUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
+package org.apache.servicecomb.pack.omega.transaction;
```
Comment (WillemJiang) [inline omega/omega-transaction/src/test/java/org/apache/servicecomb/pack/omega/transaction/TransactionContextUtilsTest.java]: License heard.

**Comment by chanjarster** [comment]
@WillemJiang Code modified, please review

**Comment by zhfeng** [comment]
@chanjarster it looks good to me. And can you rebase and push the branch again ? I think @WillemJiang had updated some changes to make the Travis CI work. We need to make sure the CI building is OK. Thanks a lot for your contributions.

**Comment by zhfeng** [comment]
The CI failed with the Akka spring tests and I think it could be unrelated to these changes.

**Comment by coolbeevip** [comment]
> The CI failed with the Akka spring tests and I think it could be unrelated to these changes.

Let me try to restart the build

**Comment by coolbeevip** [comment]
> The CI failed with the Akka spring tests and I think it could be unrelated to these changes.

I found an error message about byteman. Can you help me? 
This error does not appear every time.
```
11:06:33.494 [main] INFO  org.apache.servicecomb.pack.PackStepdefs - Cleaning up services
11:06:33.524 [main] WARN  org.apache.servicecomb.pack.PackStbooking> TrepdansformListenefs -er() : Fail to delete the by handlteman rules java.laning connection on port 9091
g.Exception: The remote byteman agent reported an error:
ERROR No rule scripts to remove


Failed scenarios:
src/test/resources/pack_booking_exception_after_post_car_compensated_scenario.feature:18 # Booking Exception After Car Hotel Secenario
```

**Comment by zhfeng** [comment]
@coolbeevip sure, please raise a JIRA issue and feel free to assign to me.

**Comment by coolbeevip** [comment]
> @coolbeevip sure, please raise a JIRA issue and feel free to assign to me.

I created an issue https://issues.apache.org/jira/browse/SCB-1397

**Pull request merged**